packages/core/src/baseclient.ts

```diff
@@ -764,10 +768,13 @@ export abstract class BaseClient<O extends ClientOptions> implements Client<O> {
 }
 
 /**
- * Verifies that return value of configured `beforeSend` is of expected type.
+ * Verifies that return value of configured `beforeSend` or `beforeSendTransaction` is of expected type.
  */
-function _ensureBeforeSendRv(rv: PromiseLike<Event | null> | Event | null): PromiseLike<Event | null> | Event | null {
-  const nullErr = '`beforeSend` method has to return `null` or a valid event.';
+function _ensureBeforeSendRv(
+  rv: PromiseLike<Event | null> | Event | null,
+  beforeSendProcessorName: string,
```

Comment:
l: does it make sense to narrow this type to `beforeSendProcessorName: 'beforeSend' | 'beforeSendTransaction'?

Reply:
Not a bad idea. Can do.